### ISSUE TITLE
Improve DataGrid binding to System.Data.DataTable

### ISSUE THREAD

**Issue by Ravatsaas** [AvaloniaUI/Avalonia#2781]
I'm playing around with Avalonia, and got stuck trying to bind a DataGrid control to a DataTable.

In WPF binding the DataGrid to the table's DefaultView property causes the table's data to display. The Avalonia DataGrid just shows the properties of the row object.

It would be a nice feature to have if it would know by default how to bind to a DataTable (or really the DataView attached to the table).

I suspect there could be a possible workaround, defining a DataTemplate, but I haven't figured it out yet. Any tips would be welcome.

**Comment by Ravatsaas** [comment]
For anyone struggling with the same problem, here's the workaround I came up with:

1. Expose the DataTable's default view on the ViewModel:
```
        public DataView DataView
        {
            get => _dataTable.DefaultView;
        }
```
2. Bind it up like this (XAML):
```
      <DataGrid Name="MyGrid" Items="{Binding DataView}">
      </DataGrid>
```
3. In the codebehind (.xaml.cs file), add a handler for the Activated event
```
        public MainWindow()
        {
            Activated += MainWindow_Activated; // <-- This line

            InitializeComponent();
#if DEBUG
            this.AttachDevTools();
#endif
        }
```
4. Implement the event handler like this:
```
        private void MainWindow_Activated(object sender, System.EventArgs e)
        {
            var grid = this.FindControl<DataGrid>("MyGrid");
            var vm = (MainWindowViewModel)DataContext;

            var cols = vm.DataView.Table.Columns;

            for (var i = 0; i < cols.Count; i++)
            {
                grid.Columns.Add(new DataGridTextColumn
                {
                    Header = cols[i].ColumnName,
                    Binding = new Binding($"Row.ItemArray[{i}]"),
                });
            }
        }
```

**Comment by hematec** [comment]
Hello,

are there any news on this topic?
When i try above suggestion, i'm not able to change the DataTable data in the DataGrid.

**Comment by FoggyFinder** [comment]
> When i try above suggestion, i'm not able to change the DataTable data in the DataGrid.

@hematec yes. I had the same issue and have to create new DataGrid each time

**Comment by qq908382818** [comment]
Binding = new Binding($"Row.ItemArray[{i}]"), it's ok.how change to axaml binding code.

**Comment by qq908382818** [comment]
DataGridTextColumn colTmp = new()
  { 
      //Binding = new Binding($"Row.ItemArray[{i}]"),
      Binding = new Binding($"Row.ItemArray[\"{dt1.Columns[i].ColumnName}\"]"), //how to work
  };
What does this code mean? Is there any explanation corresponding to DataTable or List<class> or list<object> or other type ?
